core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -2596,6 +2526,224 @@ private void parseLatestByNew(GenericLexer lexer, QueryModel model) throws SqlEx
+
+                model.addOrderBy(n, QueryModel.ORDER_DIRECTION_DESCENDING);
+                tok = SqlUtil.fetchNext(lexer);
+
+            } else {
+
+                model.addOrderBy(n, QueryModel.ORDER_DIRECTION_ASCENDING);
+
+                if (tok != null && isAscKeyword(tok)) {
+                    tok = SqlUtil.fetchNext(lexer);
+                }
+            }
+
+            if (model.getOrderBy().size() >= MAX_ORDER_BY_COLUMNS) {
+                throw err(lexer, tok, "Too many columns");
+            }
+
+        } while (tok != null && Chars.equals(tok, ','));
+        return tok;
+    }
+
+    private CharSequence parsePivot(GenericLexer lexer, QueryModel model, SqlParserCallback sqlParserCallback) throws SqlException {
+        lexer.unparseLast();
+
+        CharSequence tok;
+        expectTok(lexer, "pivot");
+
+        tok = tok(lexer, "'('");
+
+        // this corrects some issue where PIVOT is returned twice by the lexer
+        if (isPivotKeyword(tok)) {
+            optTok(lexer);
+        }
+
+        ExpressionNode expr;
+
+        while (true) {
+            expr = null;
+
+            while (expr == null && lexer.hasNext()) {
+                expr = expr(lexer, model, sqlParserCallback);
+                if (expr == null) {
+                    optTok(lexer);
+                }
+            }
+
+            if (expr == null) {
+                throw SqlException.$(lexer.lastTokenPosition(), "missing aggregate expression");
+            }
+
+            if (expr.type != ExpressionNode.FUNCTION) {
+                throw SqlException.$(lexer.lastTokenPosition(), "expected aggregate function");
+            }
+
+            CharSequence alias = null;
+
+            lexer.unparseLast();
+
+            tok = optTok(lexer);
+
+            QueryColumn col;
+
+            col = queryColumnPool.next().of(null, expr);
+
+            if (tok != null && columnAliasStop.excludes(tok)) {
+                assertNotDot(lexer, tok);
+
+                // verify that * wildcard is not aliased
+
+                if (isAsKeyword(tok)) {
+                    tok = tok(lexer, "alias");
+                    if (isAsKeyword(tok)) {
+                        tok = tok(lexer, "alias");
+                    }
+                    SqlKeywords.assertTableNameIsQuotedOrNotAKeyword(tok, lexer.lastTokenPosition());
+                    CharSequence aliasTok = GenericLexer.immutableOf(tok);
+                    validateIdentifier(lexer, aliasTok);
+                    alias = unquote(aliasTok);
+                } else {
+                    validateIdentifier(lexer, tok);
+                    SqlKeywords.assertTableNameIsQuotedOrNotAKeyword(tok, lexer.lastTokenPosition());
+                    alias = GenericLexer.immutableOf(unquote(tok));
+                }
+
+                if (col.getAst().isWildcard()) {
+                    throw err(lexer, null, "wildcard cannot have alias");
+                }
+
+                tok = optTok(lexer);
```

Comment:
we should check if  this is a "for" token and exit if it is, rather than having a hack in the expression parser